### PR TITLE
factor dft code out of sarbp.cpp

### PR DESCRIPTION
**Pull request by acolinisi**
Needed for CASPER implementation of the app. Harmless for default implementation.

```
commit e89c1c133615318718c6c0db1415cbf622e6adc0 (HEAD -> PR--factor-dft, origin/PR--factor-dft)
    dft: option to not mangle funcs

    Until CASPER compiler supports controlling the demangling
    feature of Halide when it invokes the generators.

commit a45652e9c0cdcffdad50888c4d01658f4a558098
    dft: factor further

    Needed to resolve some linking issues with the CASPER implementation of
    the app.

commit cdce62e6be2095274371d09c7ed539b70ed80f54
    factor DFT into separate file

    To be linked into the CASPER app, which does not include sarbp.cpp.
 ```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -91,7 +91,7 @@ target_link_libraries(img_output_u8_test PRIVATE Halide::Halide
                                                  ${CNPY}
                                                  img_output_u8)
 
-add_executable(sarbp_test sarbp.cpp PlatformData.cpp ImgPlane.cpp)
+add_executable(sarbp_test sarbp.cpp dft.cpp fftw.cpp PlatformData.cpp ImgPlane.cpp)
```
Comment (cimes-isi) [inline CMakeLists.txt:94]: Why two different source files?  The `fftw_plan` that we need for context is tightly coupled to the `call_dft` implementation.  Conceptually there could be need for other DFTs as well which would need their own plans, e.g., if we need to add support additional SAR functionality like autofocus.

I think we should just have `dft.h` and `dft.cpp`, or even `dft_extern.h` and `dft_extern.cpp` to provide more clarity on what these are for.  For example, I'm using a similar naming convention for `interp_extern` in a branch I still haven't merged - https://github.com/cimes-isi/halide-sar-app/tree/bp-ritsar-interp
Then the `fftw_plan` can still be local to the source file, and we add functions like `dft_ctx_init` and `dft_ctx_destroy` which manage this plan.

File: dft.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#define DLLEXPORT __declspec(dllexport)
+#else
+#define DLLEXPORT
+#endif
+
+using namespace std;
+
+#ifdef NO_MANGLE
+extern "C" DLLEXPORT
+#endif
```
Comment (cimes-isi) [inline dft.cpp:19]: With the recently added `c_plus_plus_name_mangling` feature we shouldn't need this (or the `DLLEXPORT` defined above). Why do we need the `NO_MANGLE` check?

Comment (acolinisi) [reply dft.cpp:19]: This one is easy:

    Until CASPER compiler supports controlling the demangling
    feature of Halide when it invokes the generators.
    
    No time to try to implement this in Casper compiler when we can just have the macro.

File: CMakeLists.txt (hunk 1 above)
Comment (acolinisi) [reply CMakeLists.txt]: Yes, just factoring call_dft out of main is what I started with (see commits in this PR), but then for some reason had to factor further. I just backtracked and re-tested with just the dft.cpp/h and it works fine for both default and CASPER. Re renaming -- can rename whenever.

**Comment by cimes-isi** [comment]
I think this is a lot cleaner now - thanks!

**Pull request merged**